src/debug/debugSession.ts

```diff
@@ -270,20 +270,26 @@ export class DebugSession implements IDebugSession {
         await this.startDebugging(cwd, sessionName, proxyResult.proxyDebugPort, proxyResult.proxyAppPort, async () => {
             proxyResult.proxyProcess.kill();
             if (appName) {
+                kubeChannel.showOutput("The debug session exited.");
                 await this.cleanupResource(`deployment/${appName}`);
             }
         });
     }
 
     private async cleanupResource(resourceId: string): Promise<void> {
-        kubeChannel.showOutput(`Starting to clean up debug resource...`, "Cleanup debug resource");
-        const deleteResult = await this.kubectl.invokeAsync(`delete ${resourceId}`);
-        if (deleteResult.code !== 0) {
-            kubeChannel.showOutput(`Kubectl command failed: ${deleteResult.stderr}`);
-            return;
+        const autoCleanupFlag = vscode.workspace.getConfiguration("vs-kubernetes")["vs-kubernetes.autoCleanupOnDebugTerminate"];
+        if (autoCleanupFlag) {
+            kubeChannel.showOutput(`Starting to clean up debug resource...`, "Cleanup debug resource");
+            const deleteResult = await this.kubectl.invokeAsync(`delete ${resourceId}`);
+            if (deleteResult.code !== 0) {
+                kubeChannel.showOutput(`Kubectl command failed: ${deleteResult.stderr}`);
+                return;
+            } else {
+                kubeChannel.showOutput(`Resource ${resourceId} is removed successfully.`);
+                vscode.commands.executeCommand("extension.vsKubernetesRefreshExplorer");
+            }
         } else {
-            kubeChannel.showOutput(`Resource ${resourceId} is removed successfully.`);
-            vscode.commands.executeCommand("extension.vsKubernetesRefreshExplorer");
+            kubeChannel.showOutput(`"vs-kubernetes.autoCleanupOnDebugTerminate" flag is disabled, won't clean up the resource "${resourceId}" automatically. Please do that manually.`);

```

Comment:
Suggestion: instead of this, put up a prompt saying something "Resource ${resourceId} will continue running on the cluster." with a "Clean Up" button.  If they choose Close then it leaves stuff there and they can party on it then clean it up by hand.  If they choose Clean Up then go into the standard clean up routine.

Reply:
If you felt fancy you could add an 'Always Clean Up' button which added the config entry.  Or a 'Learn More' which linked to a relevant part of the debug documentation.

Reply:
Nice, i like the suggestion. Thanks.